Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -607,34 +607,40 @@ def fromXML(self, name, attrs, content, ttFont):
 
     def __lt__(self, other):
         if type(self) != type(other):
-            return NotImplemented
+            return NotImplemented()
 
         try:
-            # implemented so that list.sort() sorts according to the spec.
             selfTuple = (
                 self.platformID,
                 self.platEncID,
                 self.langID,
                 self.nameID,
-                self.toBytes(),
             )
             otherTuple = (
                 other.platformID,
                 other.platEncID,
                 other.langID,
                 other.nameID,
-                other.toBytes(),
             )
-            return selfTuple < otherTuple
-        except (UnicodeEncodeError, AttributeError):
+        except (AttributeError):
             # This can only happen for
             # 1) an object that is not a NameRecord, or
             # 2) an unlikely incomplete NameRecord object which has not been
-            #    fully populated, or
-            # 3) when all IDs are identical but the strings can't be encoded
-            #    for their platform encoding.
-            # In all cases it is best to return NotImplemented.
-            return NotImplemented
+            #    fully populated
+            return NotImplemented()
+
+        try:
```

Comment:
instead of a new try: except block you can simply add another `except UnicodeEncodeError:` branch

Reply:
one can have as many `except SomeException:` branches as needed all in the same `try:`

Reply:
sorry I see you did intentionally as you no longer try to encode in the first try block.

Reply:
yep, also tried some `try…else…finally` approach first, but it just ended up more convoluted and hard to read.